File: src/main/java/com/uiptv/service/cache/AccountCacheReloaderFactory.java
package com.uiptv.service.cache;

import com.uiptv.util.AccountType;

public class AccountCacheReloaderFactory {
    private final AccountCacheReloader stalkerReloader = new StalkerPortalCacheReloader();
    private final AccountCacheReloader xtremeReloader = new XtremeApiCacheReloader();
    private final AccountCacheReloader m3uReloader = new M3uCacheReloader();
    private final AccountCacheReloader rssReloader = new RssCacheReloader();

    public AccountCacheReloader get(AccountType accountType) {
        if (accountType == null) {
            throw new IllegalArgumentException("Account type is required for cache reload.");
        }
        return switch (accountType) {
            case STALKER_PORTAL -> stalkerReloader;
            case XTREME_API -> xtremeReloader;
            case RSS_FEED -> rssReloader;
            case M3U8_LOCAL, M3U8_URL -> m3uReloader;
        };
    }
}

File: server/src/main/kotlin/com/uiptv/service/cache/AccountCacheReloaderFactory.kt
package com.uiptv.service.cache

import com.uiptv.service.CategoryService
import com.uiptv.service.ChannelService
import com.uiptv.service.ConfigurationService
import com.uiptv.service.HandshakeService
import com.uiptv.util.AccountType
class AccountCacheReloaderFactory @JvmOverloads constructor(
    categoryServiceProvider: () -> CategoryService = { CategoryService.getInstance() },
    configurationServiceProvider: () -> ConfigurationService = { ConfigurationService },
    handshakeServiceProvider: () -> HandshakeService = { HandshakeService.getInstance() },
    channelServiceProvider: () -> ChannelService = { ChannelService.getInstance() },
    fetchProvider: (Map<String, String>, com.uiptv.model.Account) -> String = com.uiptv.util.FetchAPI::fetch
) {
    private val stalkerReloader: AccountCacheReloader =
        StalkerPortalCacheReloader(handshakeServiceProvider, channelServiceProvider, categoryServiceProvider, configurationServiceProvider, fetchProvider)
    private val xtremeReloader: AccountCacheReloader =
        XtremeApiCacheReloader(categoryServiceProvider, configurationServiceProvider)
    private val m3uReloader: AccountCacheReloader =
        M3uCacheReloader(categoryServiceProvider, configurationServiceProvider)
    private val rssReloader: AccountCacheReloader =
        RssCacheReloader(categoryServiceProvider, configurationServiceProvider)

    fun get(accountType: AccountType?): AccountCacheReloader {
        require(accountType != null) { "Account type is required for cache reload." }
        return when (accountType) {
            AccountType.STALKER_PORTAL -> stalkerReloader
            AccountType.XTREME_API -> xtremeReloader
            AccountType.RSS_FEED -> rssReloader
            AccountType.M3U8_LOCAL, AccountType.M3U8_URL -> m3uReloader
        }
    }
}
